File: frontend/src/types/operations.ts
export interface OperationSummary {
  id: number
  timestamp: string
  file_id: number
  filename: string
  action: string
  original_name: string | null
  original_parent_id: string | null
  new_name: string | null
  new_parent_id: string | null
  confidence: number | null
  model: string | null
  reason: string | null
  status: string
  dry_run: boolean
}

File: backend/app/schemas/operations.py
from pydantic import BaseModel


class OperationSummary(BaseModel):
    id: int
    timestamp: str
    file_id: int
    filename: str
    action: str
    original_name: str | None
    original_parent_id: str | None
    new_name: str | None
    new_parent_id: str | None
    confidence: int | None
    model: str | None
    reason: str | None
    status: str
    dry_run: bool
